blueprint/core/blueprint_ScrollView.h

```diff
@@ -23,25 +23,70 @@ namespace blueprint
     class ScrollView : public View
     {
     public:
+        //==============================================================================
+        // Props following CSS Scrollbars spec with some omissions/additions.
+        // https://developer.mozilla.org/en-US/docs/Web/CSS/CSS_Scrollbars
+        static inline juce::Identifier overflowProp       = "overflow";
+        static inline juce::Identifier overflowXProp      = "overflow-x";
+        static inline juce::Identifier overflowYProp      = "overflow-y";
+        static inline juce::Identifier scrollBarWidthProp = "scrollbar-width";
+        static inline juce::Identifier scrollbarColorProp = "scrollbar-color";
+        static inline juce::Identifier scrollOnDragProp   = "scroll-on-drag";
+
         //==============================================================================
         ScrollView()
         {
             addAndMakeVisible(viewport);
-            // TODO: Set these based on props...
-            viewport.setScrollBarsShown(true, false);
+
+            //TODO: Make scrollbars hidden by default so that
+            //      explicit overflow settings required? Or show on
+            //      default and overflow: hidden can be applied?
+            viewport.setScrollBarsShown(false, false);
         }
 
         //==============================================================================
         void setProperty (const juce::Identifier& name, const juce::var& value) override
         {
             View::setProperty(name, value);
 
-            if (name == juce::Identifier("scrollbar-thumb-color"))
+            if (name == overflowYProp || name == overflowXProp || name == overflowProp)
+            {
+                const auto showVertical   = props[overflowYProp] == "scroll" || props[overflowProp] == "scroll";
+                const auto showHorizontal = props[overflowXProp] == "scroll" || props[overflowProp] == "scroll";
+
+                viewport.setScrollBarsShown(showVertical, showHorizontal);
+            }
+
+            if (name == scrollBarWidthProp)
             {
-                juce::Colour c = juce::Colour::fromString(props["scrollbar-thumb-color"].toString());
+                //TODO: Support CSS scrollbar-width "thin", "auto", "none" values
+                //      or stick with a pixel value for flexibility? Or both?
+                viewport.setScrollBarThickness(props[scrollBarWidthProp]);
+            }
+
+            if (name == scrollbarColorProp)
+            {
+                auto colors = juce::StringArray::fromTokens(props[scrollbarColorProp].toString(), " ", {});
```

Comment:
This allows a space separated pair of colors for thumb and track as per spec here: https://developer.mozilla.org/en-US/docs/Web/CSS/scrollbar-color

Reply:
Similar here too... let's have our native props be `scrollbar-thumb-color` and `scrollbar-track-color` and defer the composition by way of space separated pairs up to the js layer

Reply:
Just to make sure we're both using the same language here ... :-) 

By "native props" you mean the props C++ side? i.e. C++ holds a `scrollbarThumbColorProp` and a `scrollbarTrackColorProp`, JS `ScrollView` intercepts `scrollbar-color` changes, splits the string and passes as `scrollbar-thumb-color` and `scrollbar-track-color` to C++ ? 

So we maintain this prop name: https://developer.mozilla.org/en-US/docs/Web/CSS/scrollbar-color ?

Reply:
:+1: Obviously coming back to this comment way late, but I can see in the code change that we were on the same page!